domain/schema_validator.go

```diff
@@ -208,8 +215,51 @@ func (s *schemaValidator) Check(txnTS uint64, schemaVer int64, relatedTableIDs [
 }
 
 func (s *schemaValidator) enqueue(schemaVersion int64, relatedTableIDs []int64) {
-	s.deltaSchemaInfos = append(s.deltaSchemaInfos, deltaSchemaInfo{schemaVersion, relatedTableIDs})
-	if len(s.deltaSchemaInfos) > maxNumberOfDiffsToLoad {
+	maxCnt := int(variable.GetMaxDetalSchemaCount())
+	if maxCnt <= 0 {
+		logutil.BgLogger().Info("the schema validator enqueue", zap.Int("delta max count", maxCnt))
+		return
+	}
+
+	delta := deltaSchemaInfo{schemaVersion, relatedTableIDs}
+	if len(s.deltaSchemaInfos) == 0 {
+		s.deltaSchemaInfos = append(s.deltaSchemaInfos, delta)
+		return
+	}
+
+	lastOffset := len(s.deltaSchemaInfos) - 1
+	// The first item we needn't to merge, because we hope to cover more versions.
+	if lastOffset != 0 && unorderedEqual(s.deltaSchemaInfos[lastOffset].relatedTableIDs, delta.relatedTableIDs) {
+		s.deltaSchemaInfos[lastOffset] = delta
+	} else {
+		s.deltaSchemaInfos = append(s.deltaSchemaInfos, delta)
+	}
+
+	if len(s.deltaSchemaInfos) > maxCnt {
+		logutil.BgLogger().Info("the schema validator enqueue, queue is too long",
+			zap.Int("delta max count", maxCnt), zap.Int64("remove schema version", s.deltaSchemaInfos[0].schemaVersion))
 		s.deltaSchemaInfos = s.deltaSchemaInfos[1:]
 	}
 }
+
+func unorderedEqual(a, b []int64) bool {
```

Comment:
When `b` is a superset of `a` is ok, so the name of the function should be changed.

Reply:
This function is not correct. @AilinKid @zimulala

Reply:
Done.